### ISSUE TITLE
Updating legend line color

### ISSUE THREAD

**Issue by hanudude** [matplotlib/matplotlib#2812]
Hi,
In my Qt application, figure has a plot with multiple lines and also added navigation toolbar. In navigation toolbar last button allows us to "Edit Curves and Parameters" and there we can change the color of curves/lines. After we change it I need legend also to show the changed color and update its entry thereby.

How do I update my legend if I make any changes using navigation toolbar. Please share your suggestions. 
Thanks,
Haney.

**Comment by srikanth-narayanan** [comment]
Hi,

Why are the legend artist decoupled from the figure artist?
If legend is present should updating legend automatically part of the "Edit Curves and Parameters"?

I faced the same issue last year during one of my projects. I wrote a small callback function and linked it to home button. So whenever the user pressed the home button after updating the marker type or line type followed by a home button click the legend gets updated.

```python
def home_callback():
    print "home called"
    ax.legend()

fm = plt.get_current_fig_manager()
fm.toolbar.actions()[0].triggered.connect(home_callback)
'''
